recipes/protobuf/all/patches/protobuf-3.11.4.patch

```diff
@@ -200,7 +200,7 @@ index f90e525e..fd0c1f68 100644
 -      ARGS --${protobuf_generate_LANGUAGE}_out ${_dll_export_decl}${protobuf_generate_PROTOC_OUT_DIR} ${_protobuf_include_path} ${_abs_file}
 -      DEPENDS ${_abs_file} protobuf::protoc
 +      COMMAND "${CMAKE_COMMAND}"  #FIXME: use conan binary component
-+      ARGS -E env "DYLD_LIBRARY_PATH=${Protobuf_LIB_DIRS}:${CONAN_LIB_DIRS}:${Protobuf_LIB_DIRS_RELEASE}:${Protobuf_LIB_DIRS_DEBUG}:${Protobuf_LIB_DIRS_RELWITHDEBINFO}:${Protobuf_LIB_DIRS_MINSIZEREL}" protoc --${protobuf_generate_LANGUAGE}_out ${_dll_export_decl}${protobuf_generate_PROTOC_OUT_DIR} ${_protobuf_include_path} ${_abs_file}
++      ARGS -E env "PATH=${Protobuf_LIB_DIRS}/../bin" "DYLD_LIBRARY_PATH=${Protobuf_LIB_DIRS}:${CONAN_LIB_DIRS}:${Protobuf_LIB_DIRS_RELEASE}:${Protobuf_LIB_DIRS_DEBUG}:${Protobuf_LIB_DIRS_RELWITHDEBINFO}:${Protobuf_LIB_DIRS_MINSIZEREL}" protoc --${protobuf_generate_LANGUAGE}_out ${_dll_export_decl}${protobuf_generate_PROTOC_OUT_DIR} ${_protobuf_include_path} ${_abs_file}
```

Comment:
can't this be something like `Protobuf_BIN_DIRS` or there is no such variable?

Reply:
The `FindProtobuf` documentation suggest that there is a `Protobuf_PROTOC_EXECUTABLE` but it doesn't seem to be set.
https://cmake.org/cmake/help/latest/module/FindProtobuf.html

But I did find a `CONAN_BIN_DIRS_PROTOBUF` that looks like a cleaner option.

Reply:
This is at least partly due to fact, that we can't mimic having `Protobuf_` variables prefixes, we currently have `protobuf_`. See also #2037

Reply:
The problem is not the variable itself of casing issues. The problem is the value of the variable. Here we want to execute the `protoc` that belongs to the _build_ context, while this is a CMake file executed for a package in the _host_ context. There is only a `find_package(protobuf)` that provides everything from the _host_ context, there is no `find_package(protobuf THE_ONE_FROM_BUILD_CONTEXT)` so none of the variables that are available contain the PATH to the `protoc` from the _build_ context.

Information for `protoc` is only known by Conan (it knows the full graph) and Conan need to pass that information to CMake, unless we use some conan-convention (which is not fair), we can only populate the environment (`PATH` and `DYLIB`) and use just `protoc` inside CMake... but:
 * it requires to populate the environment: also for local build
 * we need to workaround SIP protection in Mac to use shared protoc.


If `Protobuf_PROTOC_EXECUTABLE` is a standard variable, we can populate it from Conan with the path to the proper `protoc` (still will fail for Macos because of SIP), but local-builds won't work unless the user adds this variable to the command-line command or the _toolchain_ populates it.